src/app/components/algorithms/algorithm-list/algorithm-list.component.ts

```diff
@@ -38,84 +44,140 @@ export class AlgorithmListComponent implements OnInit {
-    ];
-    this.getAlgorithms();
+    this.getAlgorithms(this.generateGetParams());
   }
 
-  getAlgorithms(): void {
-    // TODO: Fix generated services
-    /* this.algorithmService.getAlgorithms().subscribe((data) => {
-      const algorithms = data._embedded.algorithmDtoes;
-      const page = data.page;
-      console.log(algorithms);
-      console.log(page);
-    }); */
+  getAlgorithms(params: any): void {
+    this.algorithmService.getAlgorithms(params).subscribe((data) => {
+      this.prepareAlgorithmdata(JSON.parse(JSON.stringify(data)));
+    });
   }
 
-  adjustInput(): void {}
+  getAlgorithmsHateoas(url: string): void {
+    this.genericDataService.getData(url).subscribe((data) => {
+      this.prepareAlgorithmdata(data);
+    });
+  }
+
+  prepareAlgorithmdata(data): void {
+    // Clear current algorithms
+    this.algorithms = [];
+
+    // Read all incoming algorithms
+    const classicAlgs = data._embedded.classicAlgorithmDtoes;
+    const quantumAlgs = data._embedded.quantumAlgorithmDtoes;
 
-  selectionChanged(event) {
+    // Adjust Paging and Hateoas data
+    this.pagingInfo.page = data.page;
+    this.pagingInfo._links = data._links;
+
+    // Adjust Algorithm data
+    if (classicAlgs) {
+      this.algorithms = this.algorithms.concat(classicAlgs);
+    }
+    if (quantumAlgs) {
+      this.algorithms = this.algorithms.concat(quantumAlgs);
+    }
+  }
+
+  selectionChanged(event): void {
     this.selectedAlgorithms = event;
-    console.log(this.selectedAlgorithms);
   }
 
-  pageChanged(event) {
-    const newPageUrl = event;
-    console.log(newPageUrl);
+  pageChanged(event): void {
+    this.getAlgorithmsHateoas(event);
   }
 
-  dataSorted(event) {
+  dataSorted(event): void {
     this.sortData = event;
-    console.log(this.sortData);
+    this.getAlgorithms(this.generateGetParams());
   }
 
-  paginatorConfigChanged(event) {
+  paginatorConfigChanged(event): void {
     this.paginatorConfig = event;
-    console.log(this.paginatorConfig);
+    this.getAlgorithms(this.generateGetParams());
   }
 
-  deleteElements() {
-    console.log('Delete Elements: ');
-    console.log(this.selectedAlgorithms);
+  deleteElements(): void {
+    // Iterate all selected algorithms and delete them
+    for (const algorithm of this.selectedAlgorithms) {
+      this.algorithmService
+        .deleteAlgorithm(this.generateDeleteParams(algorithm.id))
+        .subscribe(() => {
+          // Refresh Algorithms after delete
+          this.getAlgorithms(this.generateGetParams());
+        });
+    }
+
+    // Clear selected algorithms
+    this.selectedAlgorithms = [];
   }
 
-  addElement() {
-    console.log('Add Element');
+  addElement(): void {
+    this.openAddAlgorithmDialog();
   }
 
-  searchElement(event) {
-    const searchText = event;
-    console.log(searchText);
+  searchElement(event): void {
+    this.searchParameter = event;
+    this.getAlgorithms(this.generateGetParams());
+  }
+
+  generateGetParams(): any {
```

Comment:
Perhaps we should move the paging / sorting stuff to the list component and add a `getParams()` there.

Reply:
Is there a easy way to use the correct method of the correct Service in the List-Component to do the sorting/paging? Like with generics? 

Or do you mean that we just generate the params in the list and return them to the parent via event and trigger the refresh of the data in the parent? In that case we would probably have to tell the List-Component how the params look like. For example for delete the algorithm needs "algoId", publication needs "publicationId", etc.

Reply:
I was thinking of the second option.
Something along the lines of:
```
interface QueryParams {
page?: number;
size?: number;
sort?: string[];
// ...
}

@Output() queryChanged = new EventEmitter<QueryParams>();
```
The listener then takes the params and appends to/overwrites them as needed:
```
onQueryChanged(params: QueryParams) {
  this.algorithmService.getAlgorithms({
    ...params,
   size: 3, // we just want 3 here, regardless of what the user says
  }).subscribe(/*...*/);
}
```

Delete/Add would be a special operation (with its own EventEmitter) that just passes the object itself to the listener. The listener then needs to make the request.
Events like `pageChange`, `paginationConfigChange`, `searchElement`, `dataSorted` could be removed though.

Reply:
Got you, we will adjust that.